File: android/app/src/main/kotlin/com/agent/aios/domain/agent/AgentStrategy.kt
package com.agent.aios.domain.agent

import com.agent.aios.domain.model.AgentResult
import com.agent.aios.domain.model.AgentStep

interface AgentStrategy {
    suspend fun execute(
        prompt: String,
        maxIterations: Int = 8,
        maxTokens: Int = 512,
        onStep: (AgentStep) -> Unit,
    ): AgentResult

    fun cancel()

    fun resolveConfirmation(approved: Boolean)

    fun initSystemPrompt()

    fun getToolManifest(): String

    fun getConversationHistory(): List<Pair<String, String>>

    fun clearHistory()
}

File: lib/domain/agent/agent_strategy.dart
import 'package:aios/domain/agent/conversation_context.dart';
import 'package:aios/domain/agent/tool_preference_tracker.dart';
import 'package:aios/domain/entities/agent_models.dart';

abstract class AgentStrategy {
  Future<AgentResult> execute(
    String prompt, {
    int maxIterations = 8,
    int maxTokens = 512,
    void Function(AgentStep)? onStep,
  });

  Future<void> warmup();

  void cancel();

  void resolveConfirmation({required bool approved});

  void resolvePermission({required bool granted});

  void setPermissionChecker(
    Future<bool> Function(String permissionKey)? checker,
  );

  String getToolManifest();

  List<({String role, String content})> getConversationHistory();

  void clearHistory();

  void setConversationContext(ConversationContext? context);

  void setToolPreferenceTracker(ToolPreferenceTracker? tracker);
}
